components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt

```diff
@@ -252,6 +264,10 @@ class Nimbus(
         return nimbus.getExperimentBranch(experimentId)
     }
 
+    @WorkerThread
+    override fun getExperimentBranches(experimentId: String): List<Branch> =
```

Comment:
@jhugman Does this need a `getExperimentBranchesOnThisThread` and similar to the other API functions and get called on the dbScope since it is calling `get_all_experiments` and thus performing a `db.read()`?

Reply:
Waiting on a response to this before landing.

Reply:
This is protected by an `@WorkerThread`; if we want to be able to call it from any/main thread, I would suggest we'd add it to the `DatabaseCache` in the Nimbus-SDK.